deltacat/exceptions.py

```diff
@@ -1,14 +1,160 @@
-class RetryableError(Exception):
-    pass
+from __future__ import annotations
+from enum import Enum
+
+
+def _pack_all_args(exception_cls, args=None, kwargs=None):
+    # This is helpful for reducing Exceptions that only accept kwargs as
+    # only positional arguments can be provided for __reduce__
+    if args is None:
+        args = ()
+    if kwargs is None:
+        kwargs = {}
+    return exception_cls(*args, **kwargs)
+
+
+class DeltaCatError(Exception):
+    msg = "A DeltaCat error occurred."
+    DEFAULT_ERROR_MESSAGE_WITH_ERRORCODE = (
+        "Error Code: {error_name}. Is Retryable Error: {is_retryable}. "
```

Comment:
This is not error code anymore nor we need to now use error name and is_retryable in the message as we have a error_name property.